toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to create and manipulate disks and partitions
+
+package configuration
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// In kickstart installation, the kickstart file will include a file (/tmp/part-include)
+// that includes the partitioning commands populated by executing the preinstall script.
+// This function aims to parse the partitioning instruction from this kickstart partition file
+// and fill up the required Disk and PartitionSetting information.
+
+// Sample partition configuration for reference: https://www.golinuxhub.com/2018/05/sample-kickstart-partition-example-raid/
+func ParseKickStartPartitionScheme(config *Config, partitionFile string) (err error) {
+
+	var (
+		parseCmd           string
+		partitionFlags     []string
+		partitionTableType PartitionTableType
+		diskInfo           map[string]int
+	)
+
+	// Check whether the config already contains partition schema
+	if len(config.Disks) > 0 && len(config.Disks[0].Partitions) > 0 {
+		logger.Log.Infof("Partition scheme already exists")
+		return
+	}
+
+	// Check whether the incoming config is for initrd
+	if config.SystemConfigs[0].Name == "ISO initrd" {
+		return
+	}
+
+	file, err := os.Open(partitionFile)
+	if err != nil {
+		logger.Log.Errorf("Failed to open file (%s)", partitionFile)
+		return
+	}
+	defer file.Close()
+
+	// Check if the file is empty
+	_, err = file.Stat()
+	if err != nil {
+		logger.Log.Errorf("Empty partition file (%s)", partitionFile)
+		return
+	}
+
```

Comment:
Parsing should be a separate subroutine from opening file so that we can mock and test its functionality.

Reply:
Has refactored the implementation